src/native/credentials_provider.c

```diff
@@ -557,9 +557,22 @@ static int s_credentials_provider_delegate_get_credentials(
     jbyteArray java_session_token =
         (*env)->GetObjectField(env, java_credentials, credentials_properties.session_token_field_id);
 
+    /**
+     * Construct Anonymous Credentials.
+     * Anonymous Credentials are used to skip signing.
+     */
+    if (java_access_key_id == NULL && java_secret_access_key == NULL) {
```

Comment:
what if someone created a credentials that have access key and secret access key as null but token is non-null?

Reply:
It is not possible because credentials are immutable and constructors do not allow that.

Reply:
`return Credentials(null, null, "something")`?

Reply:
It will fail at https://github.com/awslabs/aws-crt-java/blob/main/src/main/java/software/amazon/awssdk/crt/auth/credentials/Credentials.java#L29

Reply:
In C, aws_credentials_new will return null as well and not an object with has null,null,"something".

Reply:
oh, I see. My local repo is out of date.

Reply:
well, here, it will not go through `aws_credentials_new` at all.